### DISCUSSION TITLE
Link to job summary if present or provide URL

### DISCUSSION THREAD

**Discussion by jnewland** [community/community#123637] [Actions] [answered]
### Select Topic Area

Product Feedback

## tl;dr

Please link to Actions job summaries prominently when generated by an Actions run. Alternatively, provide the URL to these summaries to workflow authors to allow them to link to them as they see fit.

## Specific feedback

Concretely, please consider linking to Actions job summaries;

- Directly on the page displaying the logs of they job they summarize (`https://github.com/<org>/<repo>/actions/runs/<run_id>/job/<job_id>`)
  - Alternatively, the actual Markdown summary may be useful to include here!
- In the PR merge summary box instead of linking to ☝️ 

Alternatively, please make it possible for workflow authors to easily obtain the URL to the markdown summary they just generated so that they can use other GitHub features like Commit Statuses or Deployments to link to it on Pull Requests. Obtaining the Job ID for a currently executing job is currently quite challenging, requiring a series of various workarounds that make it not something that can be easily made reusable.

## Background

I'm the maintainer of an Actions workflow that my organization uses internally. I'm using the lovely [job summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) feature as a part of an attempt to provide context about the execution of the workflow to users beyond just raw logs.

Unfortunately, I'm finding that many of the frequent users of this workflow haven't ever seen the step summary that I've carefully crafted. From the context of the PR merge box, the rendered HTML generated by the summary of a step is only accessible after two clicks - "Details", then "Summary" to get from the step view back to the Actions Run overview - and a scroll to take you "below the fold" on the page. This isn't great; if a user must first _navigate through and away from the logs_ of an Actions step to get to the "summary," we're not really saving them any steps here. In fact, one user suggested we `echo` the contents of the markdown summary _in step in the workflow_ to make it easier for them to find it. 😱 

I've wasted a few hours trying to link to the job summary myself this morning, only to be disappointed by how challenging it is to obtain the last `job_id` portion of a Job summary URL:

> `https://github.com/<org>/<repo>/actions/runs/<run_id>#summary-<job_id>`

Job IDs are [notoriously hard to obtain](https://stackoverflow.com/questions/71240338/obtain-job-id-from-a-workflow-run-using-contexts); a series of workarounds are scattered around Stack Overflow and this forum. Complicating things further is the fact that the workflow I maintain is a reusable workflow called from other repos in my org; for my reusable workflow to know the _name_ that the GitHub API has assigned an actions Job so that it can obtain its ID, I must know the _name_ that the caller has used in the Job that calls my workflow. 🤯

**Comment by SimonHeimberg** [comment]
Work-Around for this case: let the summary create a (custom) link destination and link to this one.
Like `run: echo '<a name="$MY_ID"></a>' >> $GITHUB_STEP_SUMMARY` and link `https://github.com/<org>/<repo>/actions/runs/<run_id>#user-content-<MY_ID>`

I agree that gh_job_id (or whatever we call it) should be available to make many ideas easier. Also suggested in 2021 as
https://github.com/orgs/community/discussions/8945
It contains also several links to discussions

By the way, to find the current job id, `Mapping the job to a one with runner_id + status == in_progress looks reliable.`: https://github.com/orgs/community/discussions/8945#discussioncomment-12459851

**Comment by t3mi** [comment, marked as answer]
Job ID is available as `job.check_run_id`, e.g.  using `${{ format('{0}/{1}/actions/runs/{2}#summary-{3}', github.server_url, github.repository, github.run_id, job.check_run_id) }}` will get you the proper link to the summary for the running job.

**Reply by jnewland** [reply]
TIL! https://github.blog/changelog/2025-09-18-actions-yaml-anchors-and-non-public-workflow-templates/#identify-running-jobs-with-job-check-run-id-in-github-actions-contexts
